pip/req/req_install.py

```diff
@@ -195,7 +195,7 @@ def from_line(
         else:
             p, extras = _strip_extras(path)
             if (os.path.isdir(p) and
-                    (os.path.sep in name or name.startswith('.'))):
+                    (os.path.sep in name or os.path.altsep in name or name.startswith('.'))):
```

Comment:
On Unix, os.altsep might be None, you should probably modify your condition to not err in that case. If the conditional gets too long, maybe move it out to a separate statement of it's own, like [here](https://github.com/pypa/pip/blob/master/pip/resolve.py#L217).